// datalens-ui/Range/index.js
import Slider from '../Slider';
/**
 * Slider with two handles
 * @class
 * @extends Slider
 */
class Range extends Slider {
    /**
     * @param {number[]} initRange
     */
    constructor(initRange = [0, 100]) {
        super(initRange[0]);
        this.initRange = initRange;
    }
    renderInternal(el) {
        super.renderInternal(el);
        this.addClass('dl-range');
        el.innerHTML += `<input
            type="range"
            min="0"
            max="100"
            value="${this.initRange[1]}"
            autocomplete="off"
            class="dl-slider__input"
        />`;
        this.input = el.children[0];
        this.input.addEventListener('change', () => {
            this.dispatchEvent('change');
        });
        this.input2 = el.children[1];
        this.input2.addEventListener('change', () => {
            this.dispatchEvent('change');
        });
    }
    /**
     * @param {number[]} range
     */
    getValue() {
        return [
            this.input.value,
            this.input2.value
        ].sort((a, b) => (a - b));
    }
}

export default Range;
